### PR TITLE
support more jdbc operation

### PR DESCRIPTION
**Pull request by jianwan0214**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [x] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #2386

## What this PR does / why we need it:
support more jdbc operation

### CODE REVIEW

**Review by daviszhen — APPROVED** [review]

File: pkg/frontend/types.go [hunk 1]
```diff
@@ -163,3 +163,35 @@ type BackgroundExec interface {
 }
 
 var _ BackgroundExec = &BackgroundHandler{}
+
+// Column Definition Flags enum
+const (
```
Comment (fengttt) [inline pkg/frontend/types.go:168]: Always use 1 << xxx, even if small numbers.

File: pkg/frontend/mysql_protocol.go [hunk 2]
```diff
@@ -1570,6 +1578,52 @@ func (mp *MysqlProtocolImpl) sendEOFOrOkPacket(warnings, status uint16) error {
 	}
 }
 
+func setColLength(column *MysqlColumn, width int32) {
+	switch column.columnType {
+	case defines.MYSQL_TYPE_DECIMAL:
```
Comment (fengttt) [inline pkg/frontend/mysql_protocol.go:1583]: We need to have a defines.MYSQL_TYPE type and a GetLength() method.

**Review by zhangxu19830126 — APPROVED** [review]

File: pkg/frontend/types.go (hunk 1 above)
Comment (jianwan0214) [reply pkg/frontend/types.go:168]: sure

File: pkg/frontend/mysql_protocol.go (hunk 2 above)
Comment (jianwan0214) [reply pkg/frontend/mysql_protocol.go:1583]: ok

File: pkg/pb/task/task.pb.go [hunk 3]
```diff
@@ -87,13 +87,13 @@ func (ResultCode) EnumDescriptor() ([]byte, []int) {
 // TaskMetadata is a task metadata abstraction that can be scheduled for execution at any CN node.
 type TaskMetadata struct {
 	// ID task id, global unique
-	ID string `protobuf:"bytes,1,opt,name=ID,json=iD,proto3" json:"ID,omitempty"`
```
Comment (ouyuanning) [inline pkg/pb/task/task.pb.go:90]: seams you don't change task/timestamp proto file,  you can remove these pb files from commit

File: pkg/sql/plan/query_builder.go [hunk 4]
```diff
@@ -125,6 +125,7 @@ func (builder *QueryBuilder) remapAllColRefs(nodeID int32, colRefCnt map[[2]int3
 		}
 
 		for i, col := range node.TableDef.Cols {
+			col.Typ.Table = node.TableDef.Name
```
Comment (ouyuanning) [inline pkg/sql/plan/query_builder.go:128]: if Typ.autoIncr need to set in here?

File: pkg/frontend/session.go [hunk 5]
```diff
@@ -1359,12 +1359,12 @@ func (tcc *TxnCompilerContext) Resolve(dbName string, tableName string) (*plan2.
 					Width:     attr.Attr.Type.Width,
 					Precision: attr.Attr.Type.Precision,
 					Scale:     attr.Attr.Type.Scale,
+					AutoIncr:  attr.Attr.AutoIncrement,
```
Comment (ouyuanning) [inline pkg/frontend/session.go:1362]: why not TableName set here too?

File: pkg/sql/plan/query_builder.go (hunk 4 above)
Comment (ouyuanning) [reply pkg/sql/plan/query_builder.go:128]: if you you set Table & AutorIncr in Resolve interface, that will not necessary to set value here.

File: pkg/pb/task/task.pb.go (hunk 3 above)
Comment (jianwan0214) [reply pkg/pb/task/task.pb.go:90]: ok

File: pkg/sql/plan/query_builder.go (hunk 4 above)
Comment (jianwan0214) [reply pkg/sql/plan/query_builder.go:128]: ok

File: pkg/frontend/session.go (hunk 5 above)
Comment (jianwan0214) [reply pkg/frontend/session.go:1362]: sure

**Review by fengttt — APPROVED** [review]

**Review by nnsgmsone — APPROVED** [review]

**Review by ouyuanning — APPROVED** [review]

**Pull request merged**